packages/nimble-components/src/table-column/number-text/index.ts

```diff
@@ -65,6 +74,22 @@ export class TableColumnNumberText extends TableColumnTextBase {
                 return new DefaultFormatter();
         }
     }
+
+    private shouldRightAlign(): boolean {
+        if (this.alignment === NumberTextAlignment.left) {
+            return false;
+        }
+
+        if (this.alignment === NumberTextAlignment.right) {
+            return true;
+        }
+
+        // Look at format to determine the default alignment
+        if (this.format === NumberTextFormat.roundToInteger) {
```

Comment:
The thing I don't like about this is that if the data is all integral to begin with, it's unlikely a user is going to set format to `roundToInteger`, even though we probably would want to right-align in that case. I can't think of a better solution, though. Is there a strong argument not to right-align by default? When there are decimal places, it's unlikely we're going to get the decimal points to line up between rows, whether we left-align or right-align. What's the rationale for left-aligning non-integral values?

Reply:
The idea is that we only want to right-align the number if we know that the decimal separators in all values within the column will align. Since that is unlikely with the `default` format (unless as you stated, they just happen to all be integers), the decision was to left-align by default.

A `decimal` format is also coming shortly, and it will `right-align` by default since those number will always display with the same number of decimals on each row.